### PR TITLE
Implement required actions loop

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

I am adding a loop to take care of any required action so that we guarantee they pass before moving forward with the execution since any of those failing makes it almost pointless to continue for the affected user.

### CODE REVIEW

File: loadtest/control/gencontroller/controller.go
Comment (streamer45) [inline loadtest/control/gencontroller/controller.go:142]: We can review this if needed.

**Review by agnivade — APPROVED** [review]

**Comment by streamer45** [comment]
I fixed a potential issue in `joinAllTeams` as we were skipping the rest of the loop in case the member existed which could cause some entities (e.g. channels) not be loaded correctly.

**Review by agarciamontoro — COMMENTED** [review]
Changes look good, thanks!

I see this is pointing to the `separateGetUsers` branch. Why not merge this into master directly?

**Review by agarciamontoro — APPROVED** [review]

**Comment by streamer45** [comment]
> I see this is pointing to the `separateGetUsers` branch. Why not merge this into master directly?

Because changes are made on top of the others improvements Agniva worked on and they were meant to be tested together. If we merge that one before I can change the base branch or merge this one first, it's the same really.

File: loadtest/control/gencontroller/controller.go
Comment (streamer45) [reply loadtest/control/gencontroller/controller.go:242]: Sure, the return value refers to whether the action has been fully executed or not. We tend to return early in case of warnings hence the check.

**Comment by agarciamontoro** [comment]
Merged the other one. Oh, and the base branch was automatically changed. Neat.

File: loadtest/control/gencontroller/controller.go
Comment (agarciamontoro) [reply loadtest/control/gencontroller/controller.go]: Yeah, I understood it reading the code. But the code did not meet my initial expectations after having read only the signature, that's why I thought some docs would be good. Thanks!

**Comment by agnivade** [comment]
Waiting until we have the dump fully generated.

**Comment by streamer45** [comment]
@agnivade Let me know if we are happy to go ahead with this.

**Review by agnivade — APPROVED** [review]
Looks fine. Thanks

**Pull request merged**
